### PR TITLE
added basic node server

### PR DESCRIPTION
**Pull request by rhgksrua**
Added basic node server using Express web framework.

running `node server.js` or `npm start` should start the server.  Just remember to run `npm install` before running it.

`resources/` directory contains js and css for client.
`server.js` serves `index.html`.

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -15,8 +15,8 @@ module.exports = function(config) {
 
     // list of files / patterns to load in the browser
     files: [
-      './js/main/*.js',
-      './js/test/*.js'
+        './public/js/*.js',
```
Comment (JordyVlassembrouck) [inline karma.conf.js]: Following the style guide on the [wiki](https://github.com/DanyDin1801/Product-Portal/wiki), javascript files should be indented with two spaces.

File: public/js/index.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
 function displayMessage() {
     return 'Hello, World!';
```
Comment (JordyVlassembrouck) [inline public/js/index.js]: I know this method was written by me, but that was before we had the style guide.
Could you correct the indentation please?

File: server.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+var express = require('express');
+
+var app = express();
+
+var port = process.env.PORT || 3000;
+
+app.use(express.static('public'));
+
+app.get('/', function(req, res) {
+    res.sendFile(__dirname + '/html/index.html');
```
Comment (JordyVlassembrouck) [inline server.js]: 2 spaces indentation rule ;)

File: server.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+});
+
+app.listen(port, function() {
+    console.log('http://localhost:' + port);
```
Comment (JordyVlassembrouck) [inline server.js]: Same here, 2 spaces.

File: test/index_spec.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+describe('index', function() {
+  it('should return "hello world"', function() {
+      expect(displayMessage()).toBe('Hello, World!');
```
Comment (JordyVlassembrouck) [inline test/index_spec.js]: And here. Correct indentation please.

**Comment by JordyVlassembrouck** [comment]
@vinaypuppal Good point. But maybe we should do that in a separate issue.
I have seen alot of wrong indentations, so maybe we can create an issue where we implement the nodemon module and a JavaScript linter module, all within the same issue, sounds cool?

**Comment by rhgksrua** [comment]
I've fixed all indentation to 2 spaces and added `npm start` as another option to start the server.  I'll leave out `nodemon` for now.  I've read the chat and we'll figure out if we'll use `nodemon` or some kind of live reloading.  @vinaypuppal, can you create an issue for that?

**Comment by JordyVlassembrouck** [comment]
You have my approval, @rhgksrua

**Comment by vinaypuppal** [comment]
@rhgksrua done created a new issue #6

**Pull request merged**